client/my-sites/stats/controller.jsx

```diff
@@ -111,30 +109,26 @@ export default {
 	},
 
 	redirectToDefaultSitePage: function( context, next ) {
-		const siteFragment = route.getSiteFragment( context.path );
+		const siteFragment = getSiteFragment( context.path );
 
 		if ( siteFragment ) {
 			// if we are redirecting we need to retain our intended layout-focus
 			const currentLayoutFocus = getCurrentLayoutFocus( context.store.getState() );
 			context.store.dispatch( setNextLayoutFocus( currentLayoutFocus ) );
-			page.redirect( route.getStatsDefaultSitePage( siteFragment ) );
+			page.redirect( getStatsDefaultSitePage( siteFragment ) );
 		} else {
 			next();
 		}
 	},
 
 	insights: function( context, next ) {
 		const FollowList = require( 'lib/follow-list' );
-		let siteId = context.params.site_id;
-		const basePath = route.sectionify( context.path );
+		const basePath = sectionify( context.path );
 		const followList = new FollowList();
 
 		// FIXME: Auto-converted from the Flux setTitle action. Please use <DocumentHead> instead.
 		context.store.dispatch( setTitle( i18n.translate( 'Stats', { textOnly: true } ) ) );
 
-		const site = getSite( context.store.getState(), siteId );
```

Comment:
why is this change here

Reply:
I believe that it's because `siteId` wasn't being used, thus `site` wasn't needed